api/receiver/receive_results.go

```diff
@@ -74,30 +73,43 @@ func HandleResultsUpload(a API, w http.ResponseWriter, r *http.Request) {
 	}
 
 	log := shared.GetLogger(a.Context())
-	var results int
-	var getFile func(i int) (io.ReadCloser, error)
+	var results, screenshots int
+	var getResult, getScreenshot func(i int) (io.ReadCloser, error)
 	if r.MultipartForm != nil && r.MultipartForm.File != nil && len(r.MultipartForm.File["result_file"]) > 0 {
 		// result_file[] payload
 		files := r.MultipartForm.File["result_file"]
-		log.Debugf("Found %v multipart form files", len(files))
 		results = len(files)
-		getFile = func(i int) (io.ReadCloser, error) {
+		sFiles := r.MultipartForm.File["screenshot_file"]
+		screenshots = len(sFiles)
+		log.Debugf("Found %d result files, %d screenshot files", results, screenshots)
+
+		getResult = func(i int) (io.ReadCloser, error) {
 			return files[i].Open()
 		}
+		getScreenshot = func(i int) (io.ReadCloser, error) {
+			return sFiles[i].Open()
+		}
 	} else {
 		// result_url payload
 		urls := r.PostForm["result_url"]
 		results = len(urls)
-		log.Debugf("Found %v urls", results)
-		getFile = func(i int) (io.ReadCloser, error) {
+		sUrls := r.PostForm["screenshot_url"]
+		screenshots = len(sUrls)
+		log.Debugf("Found %d result URLs, %d screenshot URLs", results, screenshots)
+
+		getResult = func(i int) (io.ReadCloser, error) {
 			return fetchFile(a, urls[i])
 		}
+		getScreenshot = func(i int) (io.ReadCloser, error) {
+			return fetchFile(a, sUrls[i])
+		}
 		// Check for azure artifact URL, for unzipping.
+		// TODO(Hexcles): Support "screenshot_url" on Azure.
 		if results == 1 {
 			artifactName := getAzureArtifactName(urls[0])
 			if artifactName != "" {
 				var err error
-				results, getFile, err = handleAzureArtifact(a, artifactName, urls[0])
+				results, getResult, err = handleAzureArtifact(a, artifactName, urls[0])
```

Comment:
As discussed, we'll have to look at producing screenshots with Azure - @foolip  FYI